### PR TITLE
Avoid test flakes when creating new projects

### LINKED ISSUE #10076: flake: test/cmd/basicresources.sh oc new-project test4
Noticed in https://github.com/openshift/origin/pull/9240, full log: https://ci.openshift.redhat.com/jenkins/job/test_pull_requests_origin_check/4108/console.

```
In suite "github.com/openshift/origin/test/cmd/basicresources", test case "test/cmd/basicresources.sh:350: executing 'oc new-project test4; sleep 2; oc projects' expecting success and text 'You have one project on this server" failed:
=== BEGIN TEST CASE ===
test/cmd/basicresources.sh:350: executing 'oc new-project test4; sleep 2; oc projects' expecting success and text 'You have one project on this server: "test4".'
FAILURE after 3.774s: test/cmd/basicresources.sh:350: executing 'oc new-project test4; sleep 2; oc projects' expecting success and text 'You have one project on this server: "test4".': the output content test failed
Standard output from the command:
Now using project "test4" on server "https://127.0.0.1:28443".

You can add applications to this project with the 'new-app' command. For example, try:

    oc new-app centos/ruby-22-centos7~https://github.com/openshift/ruby-ex.git

to build a new example application in Ruby.
You are not a member of any projects. You can request a project to be created with the 'new-project' command.

There was no error output from the command.
```

### PR DESCRIPTION
**Pull request by enj**
Fixes #10076

### CODE REVIEW

**Comment by enj** [comment]
[test]

**Comment by enj** [comment]
cc @mfojtik @liggitt

**Comment by liggitt** [comment]
LGTM [merge]

**Comment by mfojtik** [comment]
LGTM thanks @enj

**Comment by soltysh** [comment]
LGTM as well.

Flake #9490.

re-[merge]

**Comment by smarterclayton** [comment]
Flake https://github.com/openshift/origin/issues/10182

**Comment by smarterclayton** [comment]
Also flake https://github.com/openshift/origin/issues/9624

**Comment by stevekuznetsov** [comment]
Give me a moment to review this before re-tagging please

**Comment by enj** [comment]
Flake on https://github.com/kubernetes/kubernetes/issues/27655
@stevekuznetsov I added the `jsonpath` checks, let me know if you want me to squash.

**Comment by enj** [comment]
[test]

File: test/cmd/projects.sh [hunk 1]
```diff
@@ -15,13 +15,18 @@ os::cmd::expect_failure_and_text 'oc projects test_arg' 'no arguments'
 os::cmd::expect_success 'oc login -u test -p test'
 os::cmd::expect_success_and_text 'oc projects' 'You are not a member of any projects'
 # add a project and expect text for a single project
-os::cmd::expect_success_and_text 'oc new-project test4; sleep 2; oc projects' 'You have one project on this server: "test4".'
-os::cmd::expect_success_and_text 'oc new-project test5; sleep 2; oc projects' 'You have access'
+os::cmd::expect_success_and_text 'oc new-project test4' 'Now using project "test4" on server '
+# avoid flakes in case of slow server update
```
Comment (stevekuznetsov) [inline test/cmd/projects.sh]: comment unnecessary

File: test/cmd/projects.sh [hunk 2]
```diff
@@ -15,13 +15,18 @@ os::cmd::expect_failure_and_text 'oc projects test_arg' 'no arguments'
...
+os::cmd::try_until_text 'oc projects' 'You have one project on this server: "test4".'
+os::cmd::expect_success_and_text 'oc new-project test5' 'Now using project "test5" on server '
+os::cmd::try_until_text 'oc projects' 'You have access to the following projects and can switch between them with '
+os::cmd::expect_success_and_text 'oc projects' 'test4'
+os::cmd::expect_success_and_text 'oc projects' 'test5'
```
Comment (stevekuznetsov) [inline test/cmd/projects.sh]: why are these two useful?

File: test/cmd/projects.sh [hunk 3]
```diff
@@ -15,13 +15,18 @@ os::cmd::expect_failure_and_text 'oc projects test_arg' 'no arguments'
...
 # test --skip-config-write
 os::cmd::expect_success_and_text 'oc new-project test6 --skip-config-write' 'To switch to this project and start adding applications, use'
-os::cmd::expect_failure 'oc config view | grep "namespace: test6"'
-os::cmd::expect_success 'sleep 2; oc projects | grep test6'
+os::cmd::expect_success_and_not_text 'oc config view -o jsonpath="{.contexts[*].context.namespace}"' '\btest6\b'
```
Comment (stevekuznetsov) [inline test/cmd/projects.sh]: Why did you need the anchors?

File: test/cmd/projects.sh (hunk 1 above)
Comment (enj) [inline test/cmd/projects.sh]: I disagree, but fine.

File: test/cmd/projects.sh (hunk 2 above)
Comment (enj) [inline test/cmd/projects.sh]: We need to know both projects exist.

File: test/cmd/projects.sh (hunk 3 above)
Comment (enj) [inline test/cmd/projects.sh]: `{.contexts[*].context.namespace}` gives you all the namespaces with a space in between them.  I am making sure that we don't match `test5 abctest6` while still matching `test5 test6`.

File: test/cmd/projects.sh (hunk 2 above)
Comment (stevekuznetsov) [inline test/cmd/projects.sh]: Won't the `os::cmd::expect_success_and_text 'oc new-project test5' 'Now using project "test5" on server '` block ensure that already?

File: test/cmd/projects.sh (hunk 1 above)
Comment (stevekuznetsov) [inline test/cmd/projects.sh]: Polling instead of `sleep` is "my scripts are running on some remote server 101" so while I agree that it's at least somewhat useful information, if we put it every single time we made the decision to poll instead of wait we'd double the length of our tests

**Comment by enj** [comment]
@stevekuznetsov Squashed and ready for merge.

**Comment by stevekuznetsov** [comment]
[merge]
